lite/examples/sound_classification/ios/SoundClassification/GMLAudioRecord/GMLAudioRecord.m

```diff
@@ -0,0 +1,185 @@
+// Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GMLAudioRecord.h"
+#import "GMLAudioError.h"
+#import "GMLRingBuffer.h"
+#import "GMLUtils.h"
+
+#define SUPPORTED_CHANNEL_COUNT 1
```

Comment:
We want to support multiple channels model. Both Python and iOS APIs support that.

Reply:
The channels should be interleaved in a flat array. I think that's the way the task library expects the data to be and Java and python works this way. 

Also the supported channel count is only internal to GMLAudioRecord. GMLAudio still supports multiple channels. If it's a file source, then we don't use the AudioRecord.

The problem is that currently the only format supported by AVAudioEngine is mono when we request audio samples. iOS 14 upwards there maybe a way to manipulate the avaudio session to connect to [stereo](https://developer.apple.com/documentation/avfaudio/avaudiosession/capturing_stereo_audio_from_built-in_microphones?language=objc). But I am not sure if after doing this it will let us use the audio engine for tapping the stereo mic to get the samples. 
We may have to create a full audio library using low level C APIs in iOS to do this.

Reply:
> The channels should be interleaved in a flat array. I think that's the way the task library expects the data to be and Java and python works this way.

Yes that's correct.

> The problem is that currently the only format supported by AVAudioEngine is mono when we request audio samples. iOS 14 upwards there maybe a way to manipulate the avaudio session to connect to [stereo](https://developer.apple.com/documentation/avfaudio/avaudiosession/capturing_stereo_audio_from_built-in_microphones?language=objc). But I am not sure if after doing this it will let us use the audio engine for tapping the stereo mic to get the samples.
We may have to create a full audio library using low level C APIs in iOS to do this.

Can you implement it this way to make it consistent with the Python and Android API?
* No need to support 3 or more channels. https://github.com/tensorflow/tflite-support/blob/d6829295cd04284f219e6b08594c699fc9abcf1a/tensorflow_lite_support/java/src/java/org/tensorflow/lite/task/audio/classifier/AudioClassifier.java#L447
* For iOS 13 and lower: Record in mono and duplicate the same content across the two channels if the model requires 2 channels.
* For iOS 14 and later: Record in stero.